### PR TITLE
Update documentation for list

### LINKED ISSUE #432: [list] Update README with added functionality
from PR #415

### PR DESCRIPTION
**Pull request by kostas-kou**
**Related issue(s) and PR(s)**  
This PR closes #432.

**Description**
Documentation was added for listing accessible datasets and their files

**How to test**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,9 +181,13 @@ will first encrypt all files in `<folder_to_upload_with_unencrypted_data>` and t
 - If the flag `--force-overwrite` is used, the tool will overwrite any already existing file.
 - The cli will exit if the input has any un-encrypred files. To override that, use the flag `--force-unencrypted`.
 
-## List files
+## List
 
-The uploaded files can be listed using the `list` parameter. This feature returns all the files in the user's bucket recursively and can be executed using:
+All the below cases require a configuration file to be downloaded before starting the operation and the `list` parameter should be used.
```
Comment (nanjiangshu) [inline README.md:186]:
```suggestion
All the following cases require a configuration file to be downloaded before starting the operation, and the `list` parameter should be used.
```

File: README.md [hunk 2]
```diff
@@ -198,6 +202,25 @@ It also allows for requesting files/filepaths with a specified prefix using:
 This command will return any file/path starting with the defined `<prefix>`.
 If no config is given by the user, the tool will look for a previous login from the user.
 
+### List datasets and their files
+
+To list datasets or files within a dataset to which the user has access, the `--datasets` flag should be used and the login URL must be provided:
```
Comment (nanjiangshu) [inline README.md:207]:
```suggestion
To list datasets or the files within a dataset that the user has access to, the `--datasets` flag should be used and the login URL must be provided:
```

File: README.md [hunk 3]
```diff
@@ -198,6 +202,25 @@ It also allows for requesting files/filepaths with a specified prefix using:
...
+
+```bash
+./sda-cli list -config <configuration_file> --datasets --url <login_url>
+```
+
+where the `<login_url>` is the url where user go for authentication. This command returns a list of accessible datasets with the number
```
Comment (nanjiangshu) [inline README.md:213]:
```suggestion
where `<login_url>` is the URL where the user goes for authentication. This command returns a list of accessible datasets, including the number
```

File: README.md [hunk 4]
```diff
@@ -198,6 +202,25 @@ It also allows for requesting files/filepaths with a specified prefix using:
...
+of files in each dataset and the total size of it.
+
+To list the files within a particular dataset, the user must provide the dataset ID (which can be obtained by executing the previously mentioned command):
```
Comment (nanjiangshu) [inline README.md:216]:
```suggestion
To list the files within a particular dataset, the user must provide the dataset ID (which can be obtained by executing the previous command):
```

File: README.md [hunk 5]
```diff
@@ -198,6 +202,25 @@ It also allows for requesting files/filepaths with a specified prefix using:
...
+
+```bash
+./sda-cli list -config <configuration_file> -dataset <datasetID> -url <login_url>
+```
+
+This command returns a list of files of the dataset with their file IDs, file sizes and file paths.
```
Comment (nanjiangshu) [inline README.md:222]:
```suggestion
This command returns a list of files within the dataset, including their file IDs, file sizes and file paths.
```

**Review by nanjiangshu — COMMENTED** [review]
Looks good. Just added a few minor comments. 

When testing, I failed to upload a file to the bigpicture with the following command:

```
sda-cli upload -config s3cmd-bp.conf -encrypt-with-key c4gh.pub.pem t1.txt
```
The error message is strange and I have never seen before

```
FATA[0014] SerializationError: failed to unmarshal error message
	status code: 502, request id: , host id: 
caused by: UnmarshalError: failed to unmarshal error message
	00000000  3c 68 74 6d 6c 3e 0d 0a  3c 68 65 61 64 3e 3c 74  |<html>..<head><t|
00000010  69 74 6c 65 3e 35 30 32  20 42 61 64 20 47 61 74  |itle>502 Bad Gat|
00000020  65 77 61 79 3c 2f 74 69  74 6c 65 3e 3c 2f 68 65  |eway</title></he|
00000030  61 64 3e 0d 0a 3c 62 6f  64 79 3e 0d 0a 3c 63 65  |ad>..<body>..<ce|
00000040  6e 74 65 72 3e 3c 68 31  3e 35 30 32 20 42 61 64  |nter><h1>502 Bad|
00000050  20 47 61 74 65 77 61 79  3c 2f 68 31 3e 3c 2f 63  | Gateway</h1></c|
00000060  65 6e 74 65 72 3e 0d 0a  3c 68 72 3e 3c 63 65 6e  |enter>..<hr><cen|
00000070  74 65 72 3e 6e 67 69 6e  78 3c 2f 63 65 6e 74 65  |ter>nginx</cente|
00000080  72 3e 0d 0a 3c 2f 62 6f  64 79 3e 0d 0a 3c 2f 68  |r>..</body>..</h|
00000090  74 6d 6c 3e 0d 0a         
```

I hope this is related to safespring storage migration, or?

File: README.md (hunk 1 above)
Comment (MalinAhlberg) [inline README.md:186]: Continuing on Nanjiang's suggestion:
```suggestion
All the following cases require a [configuration file to be downloaded](https://github.com/NBISweden/sda-cli/blob/documentation/datset-list/README.md#download-the-configuration-file) before starting the operation, and the `list` parameter should be used.
```

**Review by MalinAhlberg — COMMENTED** [review]
Nice @kostas-kou !
Also good suggestions from @nanjiangshu! I left one suggestion, and also wonder whether `-bytes` should be mentioned?

File: README.md [hunk 6]
```diff
@@ -181,9 +181,13 @@ will first encrypt all files in `<folder_to_upload_with_unencrypted_data>` and t
...
 
-## List files
+## List
 
-The uploaded files can be listed using the `list` parameter. This feature returns all the files in the user's bucket recursively and can be executed using:
+All the below cases require a configuration file to be downloaded before starting the operation and the `list` parameter should be used.
+
+### List uploaded files
+
+This feature returns all the files in the user's bucket recursively and can be executed using:
```
Comment (aaperis) [inline README.md:190]:
```suggestion
This feature returns all the files in the user's inbox recursively and can be executed using:
```

File: README.md (hunk 1 above)
Comment (kostas-kou) [reply README.md]: Done

**Comment by kostas-kou** [comment]
The information about the --bytes flag is added as well @MalinAhlberg

**Review by MalinAhlberg — APPROVED** [review]

**Review by aaperis — APPROVED** [review]

**Pull request merged**